src/harness/typeWriter.ts

```diff
@@ -21,6 +21,24 @@ namespace Harness {
         type?: string;
     }
 
+    function* forEachASTNode(node: ts.Node) {
+        const work = [[node]];

```

Comment:
You might be able to do this with just an array of `Node`.
Just push the children onto `work` in reverse order to maintain pre-order traversal in the DFS